src/lib.rs

```diff
@@ -725,6 +741,30 @@ mod test {
         assert_eq!(1, info.blocks);
     }
 
+    #[test]
+    fn test_get_cookie_user_and_pass() {
```

Comment:
I am not convinced of replicating the logic of the tested function in the test, It would probably be better to refactor out a function `fn parse_cookie(content: String) -> anyhow::Result<(Option<String>, Option<String>)>` and test it with static values?

Reply:
I agree, but I'm not sure if it another fn would be necessary. (Although I've added it now :) )

Another approach that could be used (I've updated it) would be to overwrite the .cookie file with static values and test against it, wdyt ?

Reply:
At this point maybe using a tempfile and a mocked up `ConnectParams` would have spared us to launch the `bitcoind` daemon... However, it's fine for me like it is and I am going to merge soon unless you prefer my suggestion

Reply:
I guess it can be as it's, as it's testing against a real `bitcoind` daemon created `.cookie` file on the default path 🤔